src/arrays/constructors.jl

```diff
@@ -16,4 +16,16 @@
     one_at_ind!(arr, i) = setindex!(arr, one(eltype(arr)), i)
     single_coeff(i, lens...) = one_at_ind!(zeros(lens), i)
 
-export statevec
\ No newline at end of file
+function coherentstatevec(n::Int,alpha::Number;method="direct")
+    s = rand(n)
+    z = zeros(QuArray(s))
+    for i in [1:n]
+        z = z + ((alpha^(i-1))/sqrt(factorial(i)))*statevec(i, FiniteBasis(n))
```

Comment:
Yeah I just noticed the factorial(i-1), my bad. Thanks for this ... will implement using the recursive technique.